### ISSUE TITLE
Resolver is in the wrong stack

### ISSUE THREAD

**Issue by chris-merapar** [aws-amplify/amplify-cli#1569]
**Describe the bug**
The resolver ListPageResolver for our Page GQL type has been created by Amplify in the stack for our List GQL type. This has not caused any issues until we tried to delete the List type from the schema, at which point amplify tries to move the ListPageResolver to the Page stack and this causes an error "Only one resolver is allowed per field".

**To Reproduce**
Steps to reproduce the behaviour:
1. Start with schema that includes types for Page and List
2. Deploy the schema with amplify push
3. Observe that the ListPageResolver is in the List Cloudformation stack (in the build folder locally and in the CF web console)
4. Remove the List type from the schema
5. Deploy the schema with amplify push
6. Error from Cloudformation returned on the CLI: 

UPDATE_ROLLBACK_IN_PROGRESS xxx-apihayucms-xxx-Page-xxx AWS::CloudFormation::Stack Tue Jun 04 2019 11:02:47 GMT+0100 (British Summer Time) The following resource(s) failed to create: [ListPageResolver].                                                                                                   
CREATE_FAILED               ListPageResolver                                                  AWS::AppSync::Resolver     Tue Jun 04 2019 11:02:46 GMT+0100 (British Summer Time) Only one resolver is allowed per field. (Service: AWSAppSync; Status Code: 400; Error Code: BadRequestException; Request ID: xxx)

**Expected behaviour**
The ListPageResolver should be in the Page Cloudformation stack

**Comment by kaustavghosh06** [comment]
@chris-merapar Could you share your schema (or a similar schema), so that we can reproduce this error?

**Comment by kaustavghosh06** [comment]
Closing this issue due to lack of user response. Please comment on this thread if you're still stuck and we'l re-open the issue.

**Comment by chris-merapar** [comment]
This came back to bite us again as we had to update the API when doing an amplify CLI major upgrade. I found a workaround though, reproducing here in case it helps someone else.

Since the CustomResources nested stack is always deployed after the others, we can tell Amplify not to generate the listPages resolver in the schema.graphql file, then define this resolver ourselves in the CustomResources stack. Amplify won’t deploy the new listPages resolver until the List stack has finished removing the wrongly assigned listPages resolver, so we don’t see the failure due to a duplicate resolver assignment anymore. Once this has been deployed to all environments, the workaround could be reverted and the listPages resolver should be correctly created again in the Page stack.

FYI the original schema was something like this:

`type Page @model {
  id: ID!
  title: String!
}

type List @model {
  id: ID!
  title: String!
}`

I think a very old version of Amplify put the listPages resolver in the List stack as it did a simple string match. The later versions of Amplify don't have this behaviour, so I don't think this can happen again in the future.
